### DISCUSSION TITLE
dagster._core.errors.DagsterInvariantViolationError: Cannot access partition_key for a non-partitioned run

### DISCUSSION THREAD

**Discussion by yasirad29** [dagster-io/dagster#24434] [Q&A] [answered]
I am trying to run a schedule which works with partitions, similar to the Dagster University Essentials project. When I materialize my asset and select partitions it runs successfully, However, when I test the schedule and launch the job,  I getthis error dagster._core.errors.DagsterInvariantViolationError: Cannot access partition_key for a non-partitioned run

This happens for the Dagster project when I clone it and follow the instructions as well. How do I fix this error?

**Comment by garethbrickman** [comment]
Could you please share code snippets for your schedule and job? What version of Dagster are you using?

**Reply by yasirad29** [reply]
My job file is as below

```python
from dagster import AssetSelection, define_asset_job
from ..partitions import daily_partition

members_update_job = define_asset_job(
    name="members_update_job",
    partitions_def= daily_partition, # partitions added here
    selection= AssetSelection.all()
)
```
My schedule is as below 

```python
from dagster import ScheduleDefinition
from ..jobs import members_update_job

members_update_schedule = ScheduleDefinition(
    job= members_update_job,
    cron_schedule="0 0 * * *", # every day at 12:00 am 
)
```
I am using dagster, version 1.8.6

**Comment by yasirad29** [comment, marked as answer]
I managed to fix this issue by changing the schedule using code I found on another answer here :)

```python
@schedule(job=members_update_job, cron_schedule="0 0 * * *")
def members_update_schedule(context: ScheduleEvaluationContext):
    """Update our daily assets as frequently as every 5 minutes"""
    day_to_process = context.scheduled_execution_time.strftime("%Y-%m-%d")
    return RunRequest(run_key=None, partition_key=day_to_process)  # optional
```
